toolchain/sem_ir/generic.h

```diff
@@ -0,0 +1,31 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_SEM_IR_GENERIC_H_
+#define CARBON_TOOLCHAIN_SEM_IR_GENERIC_H_
+
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::SemIR {
+
+// Information for a generic entity, such as a generic class, a generic
+// interface, or generic function.
+struct Generic : public Printable<Generic> {
+  auto Print(llvm::raw_ostream& out) const -> void {
+    out << "{decl: " << decl_id << ", bindings: " << bindings_id << "}";
+  }
+
+  // The following members always have values, and do not change throughout the
+  // lifetime of the generic.
+
+  // The first declaration of the generic entity.
+  InstId decl_id;
+  // A block containing the IDs of compile time bindings in this generic scope.
+  // The index in this block will match the `bind_index` of the instruction.
```

Comment:
I'm a little uncertain here, we only have `bind_index` on BindNameInfo, is that what you mean here?

```suggestion
  // In this block, each instruction's index will match its BindNameInfo's
  // `bind_index`.
```

Reply:
Done.